vizro-core/tests/component_library/test_component_library.py

```diff
@@ -0,0 +1,93 @@
+# ruff: noqa: F403, F405
+import dash_bootstrap_components as dbc
+import pandas as pd
+import pytest
+from dash import Dash, html
+from tests.helpers.common import compare_images
+
+from vizro.figures.library import kpi_card, kpi_card_reference
+
+df_kpi = pd.DataFrame(
+    {
+        "Actual": [100, 200, 700],
+        "Reference": [100, 300, 500],
+        "Category": ["A", "B", "C"],
+    }
+)
+
+example_cards = [
+    kpi_card(data_frame=df_kpi, value_column="Actual", title="KPI with value"),
+    kpi_card(
+        data_frame=df_kpi,
+        value_column="Actual",
+        title="KPI with aggregation",
+        agg_func="median",
+    ),
+    kpi_card(
+        data_frame=df_kpi,
+        value_column="Actual",
+        title="KPI formatted",
+        value_format="${value:.2f}",
+    ),
+    kpi_card(
+        data_frame=df_kpi,
+        value_column="Actual",
+        title="KPI with icon",
+        icon="shopping_cart",
+    ),
+]
+
+example_reference_cards = [
+    kpi_card_reference(
+        data_frame=df_kpi,
+        value_column="Actual",
+        reference_column="Reference",
+        title="KPI ref. (pos)",
+    ),
+    kpi_card_reference(
+        data_frame=df_kpi,
+        value_column="Actual",
+        reference_column="Reference",
+        agg_func="median",
+        title="KPI ref. (neg)",
+    ),
+    kpi_card_reference(
+        data_frame=df_kpi,
+        value_column="Actual",
+        reference_column="Reference",
+        title="KPI ref. formatted",
+        value_format="{value}€",
+        reference_format="{delta}€ vs. last year ({reference}€)",
+    ),
+    kpi_card_reference(
+        data_frame=df_kpi,
+        value_column="Actual",
+        reference_column="Reference",
+        title="KPI ref. with icon",
+        icon="shopping_cart",
+    ),
+]
+
+
+@pytest.mark.filterwarnings("ignore:HTTPResponse.getheader():DeprecationWarning")
+@pytest.mark.filterwarnings("ignore::pytest.PytestUnhandledThreadExceptionWarning")
+@pytest.mark.filterwarnings("ignore:unclosed file:ResourceWarning")
```

Comment:
These look the same as our ignores in `test_examples`. What is it exactly that provokes them? Is it using `dash_duo`?

Fine to leave here for now, but if we're going to be ignoring them in lots of places then let's come up with a better way to do so. rather than copying and pasting the same three filters every time.

Are lots of your other tests also going to use `dash_duo` or just these?

Reply:
As I can see for now, only this error should be reused across multiple tests:
`@pytest.mark.filterwarnings("ignore:HTTPResponse.getheader():DeprecationWarning")`

I moved it to `pyproject.toml` both from this file and integration tests

Reply:
Cool, thanks for doing this - looks good.